agent/agentssh/agentssh.go

```diff
@@ -670,17 +670,63 @@ func (s *Server) sftpHandler(logger slog.Logger, session ssh.Session) {
 	_ = session.Exit(1)
 }
 
+// CreateCommandDeps encapsulates external information required by CreateCommand.
+type CreateCommandDeps interface {
+	// CurrentUser returns the current user.
+	CurrentUser() (*user.User, error)
+	// Environ returns the environment variables of the current process.
+	Environ() []string
+	// UserHomeDir returns the home directory of the current user.
+	UserHomeDir() (string, error)
+	// UserShell returns the shell of the given user.
+	UserShell(username string) (string, error)
```

Comment:
I feel like all these methods are related to the environment, figuring out the user, env, home and shell, so EnvInfo sounds like a pretty good name to me. Deps is a bit vague IMO and makes me think of dependency injection. 😄 

---

We could also simplify this interface somewhat. All we really need are:

- `User()`
- `Environ()`

We'd just make sure the returned user has the correct home directory and shell populated. Not sure if that'd be worth the lift, though.

Reply:
`EnvInfoer` it is! 

I'd prefer to do the bare minimum here, but we can consolidate + simplify down the line!